src/Microsoft.AspNet.Mvc.ModelBinding/Binders/HeaderModelBinder.cs

```diff
@@ -30,7 +30,7 @@ protected override Task<ModelBindingResult> BindModelCoreAsync([NotNull] ModelBi
             var modelMetadata = bindingContext.ModelMetadata;
 
             // Property name can be null if the model metadata represents a type (rather than a property or parameter).
-            var headerName = modelMetadata.BinderModelName ?? modelMetadata.PropertyName ?? bindingContext.ModelName;
+            var headerName = bindingContext.BinderModelName ?? modelMetadata.PropertyName ?? bindingContext.ModelName;
```

Comment:
nope we do not have a property which combines binderModelName, propertyName and modelName. 
This is still required to do at call sites like we do in collection model binder, complexdtomodeldtobinder etc.